modules/caddyhttp/encode/zstd/zstd.go

```diff
@@ -47,7 +47,8 @@ func (Zstd) AcceptEncoding() string { return "zstd" }
 
 // NewEncoder returns a new gzip writer.
 func (z Zstd) NewEncoder() encode.Encoder {
-	writer, _ := zstd.NewWriter(nil)
+	// Limit the window to 128K.
```

Comment:
Maybe this comment could be elaborated upon to explain why this is helpful?

Would it be worth making this configurable? Is there a situation where a different value would be more optional?

Reply:
@francislavoie The thing is that it both affects the server and clients. So even if your server is doing fine, clients may have a hard time.

Yes, it will likely lower compression a bit for replies >128K, but since it is also affecting client side having a server-side switch is probably not that useful anyway.

In my test of [53927 typical web files](https://files.klauspost.com/compress/webdevdata.org-2015-01-07-4GB-subset.7z) the compressed size went from 755450358 -> 762046122 bytes, which is 100.87% the size. By default the limit is 8MB which will require the client to allocate at least that much.

Reply:
Sounds good :+1:

I think clarifying the comment to say something to that effect would be helpful for future code readers. For example, maybe:

 ```suggestion
	// The default of 8MB for the window is
	// too large for many clients, so we limit
	// it to 128K to lighten their load
```